src/corona_hakab_model/medical_state.py

```diff
@@ -9,12 +9,32 @@ class MedicalState(State, ABC):
     test_willingness: float
     detectable: bool
 
-    def __init__(self, *args, **kwargs):
-        self.susceptible = kwargs.pop('susceptible')
-        self.contagiousness = kwargs.pop('contagiousness')
-        self.test_willingness = kwargs.pop('test_willingness')
-        self.detectable = kwargs.pop('detectable')
-        super().__init__(*args, **kwargs)
+    def __init__(self, **kwargs):
+        self.susceptible = kwargs.get('susceptible')
+        self.contagiousness = kwargs.get('contagiousness')
+        self.test_willingness = kwargs.get('test_willingness')
+        self.detectable = kwargs.get('detectable')
+        super().__init__(kwargs.get('name'))
```

Comment:
this isn't what i meant, you should do something along the lines of
```
def __init__(self, *args, susceptible, contagiousness , .., **kwargs):
    self.susceptible = susceptible
    self.contagiousness = contagiousness 
    ...
    super().__init__(*args, **kwargs)
```

Reply:
Oh 🤦‍♂️